docker/pulsar/Dockerfile

```diff
@@ -33,7 +33,13 @@ COPY scripts/pulsar-zookeeper-ruok.sh /pulsar/bin
 COPY scripts/watch-znode.py /pulsar/bin
 COPY scripts/install-pulsar-client.sh /pulsar/bin
 
-RUN mkdir /pulsar/data
+# In order to support running this docker image as a container on OpenShift
+# the final image needs to give the root group sufficient permission.
+# The file permissions are preserved when copying files from this builder image to the target image.
+RUN chmod -R g+w /pulsar/conf
+RUN mkdir /pulsar/data && chmod -R g+w /pulsar/data
+RUN mkdir /pulsar/logs && chmod -R g+w /pulsar/logs
+RUN chmod -R u-w /pulsar
```

Comment:
the `/pulsar` do not have write permission anymore, so it will cause functions failed to download with default functions_worker configs. Since `downloadDirectory: download/pulsar_functions` and the `/pulsar/download/pulsar_functions` cannot be created by `pulsar` user.
We should also check other folder permissions for functions as well, like `narExtractionDirectory`.

Reply:
Thanks for testing this @freeznet. I hadn't come across that error yet. We could create the `/pulsar/download/pulsar_functions` directory as part of this docker build so that we can ensure the build has the right permissions. That solution won't help anyone who has overridden `narExtractionDirectory`, though.

I would like to avoid giving the user unnecessary permissions. The permission to write to the `/pulsar` directory seems like more permission than the user needs.